### ISSUE TITLE
Fix ModalInteraction

### ISSUE THREAD

**Issue by KILLEliteMaste** [Javacord/Javacord#1046]
Currently the ModalInteraction tries to create Components the modal contains but when receiving the event, the components are only partial complete. This causes issues with non-nullable values like options in the select menu.
1. Remove `ModalInteraction#getComponents`
2. Enhance the API by method like we have for Slash Command Interactions in `SlashCommandInteractionOptionsProvider` so we can get options by index(if that makes sense) or by custom id

**Comment by AhmedA1559** [comment]
> Currently the ModalInteraction tries to create Components the modal contains but when receiving the event, the components are only partial complete. This causes issues with non-nullable values like options in the select menu.
> 2. Enhance the API by method like we have for Slash Command Interactions in `SlashCommandInteractionOptionsProvider` so we can get options by index(if that makes sense) or by custom id
Do you mean the API for the select menu specifically?

**Comment by KILLEliteMaste** [comment]
Yeah. As we do not construct complete component objects, we need component specific methods to get the values. While this unluckily worked for TextInput, this does not work for Select menus as not everything will be sent to construct them. Therefore, the existing methods for the TextInput have to be changed to not rely on the created component and instead specific methods to get the values need to be added. Same applies of course for select menus
